server/src/main/java/de/uftos/services/ConstraintInstanceService.java

```diff
@@ -177,11 +186,67 @@ public ConstraintInstance update(String signatureId, String id,
+    ConstraintInstance instance = this.getInstanceById(signatureId, id);
 
     this.repository.delete(instance);
   }
 
+  private ConstraintInstance getInstanceById(String signatureId, String id) {
+    ConstraintSignature signature = this.signatureRepository.findById(signatureId)
+        .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+
+    return this.repository.findBySignatureAndId(signature, id)
+        .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+  }
+
+  private List<ConstraintArgumentDisplayName> processConstraintInstance(
+      ConstraintInstance constraintInstance) {
+    List<ConstraintArgumentDisplayName> displayNames = new ArrayList<>();
+    for (ConstraintArgument constraintArgument : constraintInstance.getArguments()) {
+      ConstraintArgumentDisplayName displayName = getDisplayName(constraintArgument);
+      displayNames.add(displayName);
+    }
+    return displayNames;
+  }
+
+  private List<ConstraintArgumentDisplayName> processConstraintInstances(
+      List<ConstraintInstance> constraintInstances) {
+    List<ConstraintArgumentDisplayName> displayNames = new ArrayList<>();
+    for (ConstraintInstance constraintInstance : constraintInstances) {
+      displayNames.addAll(processConstraintInstance(constraintInstance));
+    }
+    return displayNames;
+  }
+
+  private ConstraintArgumentDisplayName getDisplayName(ConstraintArgument constraintArgument) {
+    String id = constraintArgument.getValue();
+    return switch (constraintArgument.getConstraintParameter().getParameterType()) {
+      case TAG ->
+          new ConstraintArgumentDisplayName(id, tagRepository.findById(id).orElseThrow().getName());
+      case ROOM -> new ConstraintArgumentDisplayName(id,
+          roomRepository.findById(id).orElseThrow().getName());
+      case GRADE -> new ConstraintArgumentDisplayName(id,
+          gradeRepository.findById(id).orElseThrow().getName());
+      case LESSON -> new ConstraintArgumentDisplayName(id,
+          lessonRepository.findById(id).orElseThrow().getYear());
+      case STUDENT -> {
+        Student student = studentRepository.findById(id).orElseThrow();
+        yield new ConstraintArgumentDisplayName(id,
+            "%s %s".formatted(student.getFirstName(), student.getLastName()));
+      }
+      case SUBJECT -> new ConstraintArgumentDisplayName(id,
+          subjectRepository.findById(id).orElseThrow().getName());
+      case TEACHER -> {
+        Teacher teacher = teacherRepository.findById(id).orElseThrow();
+        yield new ConstraintArgumentDisplayName(id,
+            "%s %s".formatted(teacher.getFirstName(), teacher.getLastName()));
+      }
+      case STUDENT_GROUP -> new ConstraintArgumentDisplayName(id,
+          studentGroupRepository.findById(id).orElseThrow().getName());
+      case TIMESLOT -> new ConstraintArgumentDisplayName(id,
+          timeslotRepository.findById(id).orElseThrow().getDay().toString());
+    };
+  }
```

Comment:
Gotta love those lol

Reply:
I tried to make the switch as tidy and simple as possible. I don't know if it's possible to improve it

Reply:
Yeah I don't think so either tbh